### PR TITLE
9 section 1 delete an item

### PR DESCRIPTION
**Pull request by megroth2**
I’m so brain dead after this story its hard to remember everything I did, but here are some highlights aside from all the obvious delete action stuff:
- added model associations and tests
- installed shoulda-matchers gem to test associations
- resolved merge conflicts (had to pull the latest changes down in order to get the create item functionality so I could test postman)
- There are some edge case tests that I commented out because I couldn’t quite get them working, was worried I was overthinking them and misunderstanding the requirements, and decided to focus on the critical pieces of functionality instead. The two tests are in item_spec.rb and items_request_spec.rb. I’ll plan to revisit if I can but leave them commented out for now.
- the postman test for delete passed!
- simplecov test coverage is at 99.75%!

### CODE REVIEW

File: Gemfile
Comment (s2an) [inline Gemfile:1]: I'm surprised this was the only additional gem we needed!

File: app/controllers/api/v1/items_controller.rb
Comment (s2an) [inline app/controllers/api/v1/items_controller.rb:1]: I think that might have happened when I was rebasing main and using VS Code's merge editor. Moving forward I will run the tests after handling the conflicts to ensure it was done correctly.

File: app/models/item.rb [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
   validates :unit_price, presence: true
-end
\ No newline at end of file
+
+  def destroy_with_invoice_items_and_invoices
+    invoices_to_destroy = Invoice.joins(:invoice_items)
+                                  .where(invoice_items: { item_id: self.id })
+                                  .group(:id)
+                                  .having('COUNT(*) = 1')
+
```
Comment (s2an) [inline app/models/item.rb:15]: Nice use of ActiveRecord.

File: config/routes.rb [hunk 2]
```diff
@@ -10,7 +10,7 @@
   namespace :api do
     namespace :v1 do
       resources :merchants, only: [:index, :show]
-      resources :items, only: [:index, :show, :create, :update]
+      resources :items, only: [:index, :show, :create, :update, :destroy]
     end
```
Comment (s2an) [inline config/routes.rb:14]: we can just use `resources :items` at this point (during the refactor)
And if we want to get fancy, we can change merchants to `except: :destroy`

File: spec/models/item_spec.rb
Comment (s2an) [inline spec/models/item_spec.rb:1]: Awesome work! And I'll get the validations added to the test during the refactor.

File: app/controllers/api/v1/items_controller.rb
Comment (megroth2) [reply app/controllers/api/v1/items_controller.rb:1]: Oh no worries - I was also working with the merge conflicts on this file so I may have introduced it!
